src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -237,18 +239,23 @@ internal override async ValueTask<QuicStreamProvider> AcceptStreamAsync(Cancella
 
             ThrowIfDisposed();
 
-            if (await _acceptQueue.Reader.WaitToReadAsync(cancellationToken))
+            MsQuicStream stream;
+
+            try
+            {
+                stream = await _acceptQueue.Reader.ReadAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (ChannelClosedException)
             {
-                if (_acceptQueue.Reader.TryRead(out MsQuicStream stream))
+                throw _abortErrorCode switch
                 {
-                    if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
-                    return stream;
-                }
+                    -1 => new QuicOperationAbortedException(), // Shutdown initiated by us.
+                    long err => new QuicConnectionAbortedException(err) // Shutdown initiated by peer.
+                };
             }
 
             if (NetEventSource.IsEnabled) NetEventSource.Exit(this);
-
-            return null;
```

Comment:
I think it's better to continue returning null than to throw an exception from AcceptStreamAsync when the connection is closing. The connection being closed is a common enough that throwing an exception feels noisy. I know this will make the QuicConnection.AcceptStreamAsync() wrapper a little more complicated, but I think it's worth it.

Reply:
Yeah, looking back on this, I agree as well.

Reply:
I think that anyone familiar with sockets will expect consistency with that, which is why I went that direction, but I also think most everyone else might expect it will return null. The question becomes who is the more likely user. It's trouble to merge Stream and Socket 🙃.

I don't feel strongly about it. We can change it.

Reply:
Let's keep it with the "current" behavior of returning null, but fix it in QuicConnection.AcceptStreamAsync.

Reply:
Actually, I think this needs some more thought now. We should consider how consistent the APIs between QuicStream, QuicConnection, QuicListener need to be in how they behave on disposal. QuicStream.ReadAsync() etc. do not return a nullable type and will need to throw an exception.

Reply:
I'm going to commit this for now to unblock things (the APIs are currently broken, returning null triggers a bug) but will open an issue to figure this out.

Reply:
Issue: https://github.com/dotnet/runtime/issues/32142